syntax = "proto3";
package properties;


message Property {
  enum Mode {
    UNKNOWN = 0;
    REPLACE = 1;
    APPEND = 2;
  }

  uint64 object_id = 1;
  uint64 type = 2;
  string value = 3;
  Mode mode = 4;
}

message PropertiesList {
  uint64 object_id = 1;
  repeated uint64 types = 2;
}

message SetPropertiesRequest {
  repeated Property properties = 1;
}

message SetPropertiesResponse {
}

message GetPropertiesRequest {
  repeated PropertiesList objects = 1;
}

message GetPropertiesResponse {
  repeated Property properties = 1;
}

message DebugClearServiceRequest {}
message DebugClearServiceResponse {}